app/src/CMakeLists.txt

```diff
@@ -0,0 +1 @@
+target_sources(app PRIVATE main.c foo.c)
```

Comment:
Doing it this way is unlike how we structure any of the built-in samples. We always have app/CMakeLists.txt, not app/src/CMakeLists.txt.

I think consistency is important here. Let's not change it for the out of tree sample or people will wonder why it's different.

Reply:
removed src CMakeLists.txt